include/crocoddyl/multibody/contacts/contact-1d.hxx

```diff
@@ -72,34 +112,62 @@ void ContactModel1DTpl<Scalar>::calcDiff(
       *state_->get_pinocchio().get(), *d->pinocchio, joint, pinocchio::LOCAL,
       d->v_partial_dq, d->a_partial_dq, d->a_partial_dv, d->a_partial_da);
   const std::size_t nv = state_->get_nv();
-  pinocchio::skew(d->vv, d->vv_skew);
-  pinocchio::skew(d->vw, d->vw_skew);
+  pinocchio::skew(d->v.linear(), d->vv_skew);
+  pinocchio::skew(d->v.angular(), d->vw_skew);
   d->fXjdv_dq.noalias() = d->fXj * d->v_partial_dq;
   d->fXjda_dq.noalias() = d->fXj * d->a_partial_dq;
   d->fXjda_dv.noalias() = d->fXj * d->a_partial_dv;
-
-  d->da0_dx.leftCols(nv).row(0).noalias() = d->fXjda_dq.row(2);
-  d->da0_dx.leftCols(nv).row(0).noalias() +=
-      d->vw_skew.row(2) * d->fXjdv_dq.template topRows<3>();
-  d->da0_dx.leftCols(nv).row(0).noalias() -=
-      d->vv_skew.row(2) * d->fXjdv_dq.template bottomRows<3>();
-
-  d->da0_dx.rightCols(nv).row(0).noalias() = d->fXjda_dv.row(2);
-  d->da0_dx.rightCols(nv).row(0).noalias() +=
-      d->vw_skew.row(2) * d->fJf.template topRows<3>();
-  d->da0_dx.rightCols(nv).row(0).noalias() -=
-      d->vv_skew.row(2) * d->fJf.template bottomRows<3>();
+  d->da0_local_dx.leftCols(nv) = d->fXjda_dq.template topRows<3>();
+  d->da0_local_dx.leftCols(nv).noalias() +=
+      d->vw_skew * d->fXjdv_dq.template topRows<3>();
+  d->da0_local_dx.leftCols(nv).noalias() -=
+      d->vv_skew * d->fXjdv_dq.template bottomRows<3>();
+  d->da0_local_dx.rightCols(nv) = d->fXjda_dv.template topRows<3>();
+  d->da0_local_dx.rightCols(nv).noalias() +=
+      d->vw_skew * d->fJf.template topRows<3>();
+  d->da0_local_dx.rightCols(nv).noalias() -=
+      d->vv_skew * d->fJf.template bottomRows<3>();
+  const Eigen::Ref<const Matrix3s> oRf = d->pinocchio->oMf[id_].rotation();
 
   if (gains_[0] != 0.) {
-    const Eigen::Ref<const Matrix3s> oRf = d->pinocchio->oMf[id_].rotation();
-    d->oRf(0, 0) = oRf(2, 2);
-    d->da0_dx.leftCols(nv).noalias() += gains_[0] * d->oRf * d->Jc;
+    pinocchio::skew(d->dp_local, d->dp_skew);
+    d->da0_local_dx.leftCols(nv).noalias() +=
+        gains_[0] * d->dp_skew * d->fJf.template bottomRows<3>();
+    d->da0_local_dx.leftCols(nv).noalias() +=
+        gains_[0] * d->fJf.template topRows<3>();
   }
   if (gains_[1] != 0.) {
-    d->da0_dx.leftCols(nv).row(0).noalias() +=
-        gains_[1] * d->fXj.row(2) * d->v_partial_dq;
-    d->da0_dx.rightCols(nv).row(0).noalias() +=
-        gains_[1] * d->fXj.row(2) * d->a_partial_da;
+    d->da0_local_dx.leftCols(nv).noalias() +=
+        gains_[1] * d->fXjdv_dq.template topRows<3>();
+    d->da0_local_dx.rightCols(nv).noalias() +=
+        gains_[1] * d->fJf.template topRows<3>();
+  }
+  switch (type_) {
+    case pinocchio::ReferenceFrame::LOCAL:
+      d->da0_dx.row(0) = d->da0_local_dx.row(2);
```

Comment:
Should we always assume constraints in Z?

It would be useful if this could be configured via a rotation matrix moving the z-axis.
This will require 
 - including this information in the constructors and
 - writing unit tests with random rotation matrix.

Could you handle this?

Reply:
Please ping me when this is done. Thanks!